python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -224,9 +224,14 @@ def find_parent_materialized_asset_partitions(
     target_asset_keys = target_asset_selection.resolve(asset_graph)
 
     for asset_key in target_asset_selection.upstream(depth=1).resolve(asset_graph):
-        records = instance_queryer.get_materialization_records(
-            asset_key=asset_key, after_cursor=latest_storage_id
-        )
+        # if the asset is non-partitioned, we only need the latest materialization
+        if not asset_graph.is_partitioned(asset_key):
+            latest_record = instance_queryer.get_latest_materialization_record(asset_key)
+            records = [latest_record] if latest_record else []
+        else:
```

Comment:
in a future PR (post-after_cursor being added to the materialization_count_by_partition function), this branch will get more efficient.

for now, if this is your first time turning on the sensor,  and you have partitioned assets, this will require querying for all historical materialization events for those assets (super expensive if you have a big history!).

however, at least for the non-partitioned assets, we'll only grab the first one, which provides a big performance boost for that (more common) case